controllers/backstage_controller.go

```diff
@@ -181,7 +181,27 @@ func (r *BackstageReconciler) applyObjects(ctx context.Context, objects []model.
 		}
 
 		if err := r.patchObject(ctx, baseObject, obj); err != nil {
-			return fmt.Errorf("failed to patch object %s: %w", obj.Object(), err)
```

Comment:
I find it more readable as it allows for a better understanding of the intent and exit conditions rather than long statements.
But also, as generally practiced, trying to keep the normal code path at a minimal indentation.
Anyway, I've tried to simplify this - see [`ec43e69` (#369)](https://github.com/janus-idp/operator/pull/369/commits/ec43e699618841fa0e68d42dd8523e6966a487a0)

Reply:
Well, TBH I am not sure that this kind of "patching" is 100% correct (indeed there is a reason to keep some SS's fields immutable?) so I 'd like this patch looks as "isolated" as possible and would not look like the main flow.
Apparently we can not avoid it for upgrading but I would consider it as a rather temporary TBH.

Reply:
I took a quick look at how more specialized DB Operators would reconcile StatefulSets, and they appear to handle this the same way. See:
- https://github.com/zalando/postgres-operator/blob/master/pkg/cluster/resources.go#L198-L238
- https://github.com/CrunchyData/postgres-operator/blob/master/internal/controller/postgrescluster/pgadmin.go#L330-L351
- https://github.com/percona/percona-postgresql-operator/blob/main/internal/controller/postgrescluster/pgadmin.go#L332-L353

So, unless I've missed something, it seems to me that would be part of the "normal" code path.

Reply:
OK
My proposal was mostly about to move this logic from to patchObject function since it is patch specific, nothing critically important.